system/lib/wasmfs/syscalls.cpp

```diff
@@ -329,6 +335,7 @@ __wasi_fd_t __syscall_open(long pathname, long flags, long mode) {
     // If O_DIRECTORY is also specified, still create a regular file:
     // https://man7.org/linux/man-pages/man2/open.2.html#BUGS
     if (flags & O_CREAT) {
+      mode &= S_IALLUGO;
```

Comment:
Why is the default to give permission to the user, group, and other? I'd expect just user, naively, but probably I'm missing something.

This does seem harmless either way, given we are a single-user environment at least for now. If it's just for now, then a comment at least would be good.

Reply:
A comment sounds good 👍